docs/visual-studio-extensibility/nuget-api-in-visual-studio.md

```diff
@@ -23,35 +41,72 @@ As of NuGet 3.3+, NuGet exports the following
-- [`IVsTemplateWizard`](#ivstemplatewizard-interface): Designed for project/item templates to include pre-installed packages; this interface is *not* meant to be invoked from code and has no public methods. (3.3+)
 
-## Using NuGet services
+#### NuGet.SolutionRestoreManager
 
-1. Install the [`NuGet.VisualStudio`](https://www.nuget.org/packages/NuGet.VisualStudio) package into your project, which contains the `NuGet.VisualStudio.dll` assembly.
+These interfaces are designed for project systems to interact with NuGet, allowing the project system to notify NuGet of changes to `PackageReference`s, and orchestrate batch updates. Visual Studio extensions that are not project systems probably will not benefit from these APIs.
+
+- [`IVsSolutionRestoreService`](#ivssolutionrestoreservice-interface) (4.0+)
+- [`IVsSolutionRestoreService2`](#ivssolutionrestoreservice2-interface) (4.3+)
+- [`IVsSolutionRestoreService3`](#ivssolutionrestoreservice3-interface) (5.1+)
+- [`IVsSolutionRestoreService4`](#ivssolutionrestoreservice4-interface) (6.0+)
+- [`IVsSolutionRestoreStatusProvider`](#ivssolutionrestorestatusprovider-interface) (6.0+)
 
-    When installed, the package automatically sets the **Embed Interop Types** property of the assembly reference to **True**. This makes your code  resilient against version changes when users update to newer versions of NuGet.
+## Using NuGet Services
 
 > [!Warning]
-> Do not use any other types besides the public interfaces in your code, and do not reference any other NuGet assemblies, including `NuGet.Core.dll`.
+> Do not use any other types besides the public interfaces in your code, and do not reference any other NuGet assemblies, such as `NuGet.Protocol.dll`, `NuGet.Frameworks.dll`, and so on.
+
+In order to maximize the backwards compatibility promises we make, but also providing ourselves the flexibility to implement new features, performance improvements, and bug fixes in Visual Studio, we do not support the NuGet Client SDK being used in Visual Studio, and we do not provide binding redirects in `devenv.exe.config` to assemblies other than our VS extensibility contracts.
+
+If you would like a new NuGet related API in Visual Studio, please search [NuGet's Home repo](https://github.com/NuGet/Home/) and up vote any existing issues if you find a similar one. If you can't find an existing feature request to up vote, please create one.
+
+### Brokered Services
+
+1. Install the [`NuGet.VisualStudio.Contracts`](https://www.nuget.org/packages/NuGet.VisualStudio.Contracts/) package into your project, as well as [`Microsoft.VisualStudio.SDK`](https://www.nuget.org/packages/Microsoft.VisualStudio.SDK).
+
+1. Use the `IAsyncServiceProvider` to get Visual Studio's service broker, and use that to get NuGet's service.
+
+   ```cs
+   // Your AsyncPackage implements IAsyncServiceProvider
+   IAsyncServiceProvider asyncServiceProvider = this;
```

Comment:
Any chance we can link to relevant docs for other products?

Reply:
No, we can't add links inside code blocks. I added a link to `AsyncPackage` above the code block, if that's what you think would be most useful, although you commented on the line with `IAsyncServiceProvider`, so maybe you wanted me to link to that interface's docs?  Or did you intend for me to link to all the types, the brokered service container and the type returned by `GetFullAccessServiceBroker`?

Reply:
Wondering if there's detail on how to detail with brokered services.

Maybe something that's in the extensibility docs.

Reply:
Maybe Mads knows?

Reply:
Following an email thread, no, unfortunately there are no extensibility docs on using brokered services, only the reference docs on the APIs. They created (internal) issues to create some docs. I'm following the issues, and once they're done I'll be able to link them.